Userland/Libraries/LibDSP/Track.cpp

```diff
@@ -64,6 +65,13 @@ bool NoteTrack::check_processor_chain_valid() const
 ErrorOr<void> Track::resize_internal_buffers_to(size_t buffer_size)
 {
     m_secondary_sample_buffer = TRY(FixedArray<Sample>::create(buffer_size));
+    bool false_variable = false;
+    while (!m_sample_lock.compare_exchange_strong(false_variable, true)) {
+        usleep(1);
+    }
+    FixedArray<Sample> cache = TRY(FixedArray<Sample>::create(buffer_size));
```

Comment:
Move this before the lock so that we hold the lock for as short as possible. (A swap is only a few instructions, this makes sure that stuttering from concurrent buffer access is as rare as feasible)